Python/Pomodoro_Timer/Pomodoro.py

```diff
@@ -0,0 +1,152 @@
+import time
+
+# path of host file in windows
+host_path = r"C:\Windows\System32\drivers\etc\hosts"
+
+# URL of websites to block
+block_list = [
+    'www.facebook.com', 'facebook.com',
+    'www.youtube.com', 'youtube.com',
+    'www.gmail.com', 'gmail.com',
+    'www.instagram.com', 'instagram.com',
+    'www.twitter.com', 'twitter.com'
+]
+
+# redirecting above URLs to this localhost to ensure blocking
+redirect = "127.0.0.1"
+
+
+def block_websites():
+    """
+    The function will open the host file and add the block-list websites to
+    the file if it is not already present and redirect it to the localhost
+    for blocking
+    """
+
+    try:
+        # Opening the host file in reading and writing mode
+        with open(host_path, 'r+') as h_file:
+            content = h_file.read()
+
+            for website in block_list:
+
+                # Website is already blocked
+                if website in content:
+                    pass
+
+                # To redirect the website to be blocked
+                else:
+                    h_file.write(redirect + "\t" + website + "\n")
+
+        return 1
+
+    except PermissionError:
+        print('\nTry running the cmd in admin mode and '
+              'then run this code to enable this functionality!')
+        return 0
+
+    except FileNotFoundError:
+        print('\nThis functionality is not supported for your OS!')
+        return 0
+
+
+def remove_websites():
+    """
+    The function will unblock the block_list websites by opening the file
+    and removing the changes we made before
+    """
+    try:
+        with open(host_path, "r+") as file:
+            content = file.readlines()
+            file.seek(0)
+            for lines in content:
+                if not any(website in lines for website in block_list):
+                    file.write(lines)
+                    # print('written')
+            file.truncate()
+
+    finally:
+        pass
+
+
+def pomodoro():
+    """
+    This function has the implementation of the user-friendly Pomodoro timer
+    along with the website blocking functionality.
+    """
+
+    print('\n------------------ POMODORO TIMER ------------------\n')
+    print('\nPomodoro timer helps to break down your work into 25 minutes of '
+          'high focus intervals separated by short breaks of 5 minutes.')
+    print('\nFor extra focus, would you like to enable website blocker? '
+          '\n**Selecting this option will block all of the popular '
+          'social media sites throughout the 25 min duration**')
+    enable = input('\nPress (y) to enable and any other key to skip: ')
+
+    cycle = 0
+
+    while 1:
+        # Feature to block social media websites
+        if enable == 'y':
+            if block_websites():
+                pass
+            else:
+                enable = ''
+
+        _ = input('\nPress any key to start the timer: ')
```

Comment:
Just a question, is using a `_` necessary? Can we use some other name? Underscores in variable naming are rather sacred in case of Python 😛